src/iroh.rs

```diff
@@ -60,14 +77,39 @@ impl VeilidIrohBlobs {
         let router = veilid.routing_context().unwrap();
         let (route_id, route_id_blob) = make_route(&veilid).await.unwrap();
 
-        return Ok(Self::new(
+        let blobs = Self::new(
             veilid,
             router,
             route_id_blob,
             route_id,
             updates,
             store,
-        ));
+            base_dir.clone(), // Pass base_dir
+        );
+    
+        // Load the collection_hashes HashMap
+        let hash_file_path = base_dir.join("collection_hashes_hash");
```

Comment:
We shouldn't track collection hashes like this. It should just be store tags so we can avoid needing to have mutextes and stuff on the veilid iroh blobs class